pkg/controllers/pod/pod_controller.go

```diff
@@ -168,18 +173,20 @@ func NewPodController(ctx context.Context, k8sClientset *kubernetes.Clientset, c
 				return
 			}
 
-			wepInterface, err := podConverter.Convert(obj)
+			wepDataList, err := podConverter.Convert(obj)
 			if err != nil {
 				log.WithError(err).Errorf("Error while converting %v to wep.", key)
 				return
 			}
 
 			// Clean up after the deleted workload endpoint.
-			wep := wepInterface.(converter.WorkloadEndpointData)
-			k := podConverter.GetKey(wep)
-			resourceCache.Clean(k)
-			workloadEndpointCache.Lock()
-			delete(workloadEndpointCache.m, k)
+			for _, wepData := range wepDataList {
+				k := podConverter.GetKey(wepData)
+				resourceCache.Clean(k)
+				workloadEndpointCache.Lock()
+				delete(workloadEndpointCache.m, k)
+			}
+
 			workloadEndpointCache.Unlock()
```

Comment:
Does this need to be in the loop with the `Lock`?

Reply:
oh boy, absolutely